File: ochinpo.py
#!/usr/bin/env python
# coding: UTF-8
# For python 2.x
import sys
import random

def main():
    str = ['お', 'ち', 'ん', 'ぽ']
    complete = 0
    cnt = 0
    rnd = 0
    while complete < 4:
        cnt += 1
        rnd = random.randint(0, 3)
        sys.stdout.write(str[rnd])
        complete = complete+1 if rnd == complete else 0
    print ""
    print cnt, "回目で完成しました"

if __name__ == '__main__':
    main()

File: ochinpo.rb
#!/usr/bin/env ruby
# -*- coding: utf-8 -*-
# for Ruby 2.x

str = ["お", "ち", "ん", "ぽ"]
complete = 0
cnt = 0
rnd = 0

while complete < 4 do
  cnt += 1
  rnd = rand(4)
  print(str[rnd].chomp)
  complete = (rnd == complete) ? complete+1 : 0
end

print("\n", cnt, "回目で完成しました\n")
